core/src/main/java/com/aem/demo/core/services/impl/RssFeedServiceImpl.java

```diff
@@ -41,4 +51,87 @@ public List<NewsCard> readFeed() {
         return feed.getEntries().stream().map(NewsCard::new)
                 .collect(Collectors.toList());
     }
+
+    @Override
+    public String saveRssFeedNodes(List<NewsCard> cardList) {
+        String nodeCreated= StringUtils.EMPTY;
+        try {
+            ResourceResolver resolver = ResolverUtil.newResolver(resourceResolverFactory);
+            Session session = resolver.adaptTo(Session.class);
+
+            if (session.nodeExists(NODE_PATH)){
+                nodeCreated = addNewsCardNode(session, cardList);
+            } else {
+                addParentNode(session);
+                nodeCreated = addNewsCardNode(session, cardList);
+            }
+
+        } catch (Exception e) {
+            log.error("\n Error while creating node - {} ",e.getMessage());
+        }
+        return nodeCreated;
+    }
+
+    private int count = 0;
+
+    private String addNewsCardNode(Session session, List<NewsCard> list){
+
+        try {
+            for (NewsCard elm:list) {
+                count++;
+                String nodeName= count + " newscard_node";
+                Node node = session.getNode(NODE_PATH);
+                Node cardNode = node.addNode(nodeName, NEWSCARDNODE_TYPE);
+                cardNode.setProperty("topic", elm.getTopic());
+                cardNode.setProperty("article", elm.getArticle());
+                cardNode.setProperty("link", elm.getLink());
+                cardNode.setProperty("pubDate", String.valueOf(elm.getPubDate()));
+                cardNode.setProperty("image", elm.getImage());
+                saveImage(cardNode, elm.getImage());
+                session.save();
+            }
+
+
+        } catch (Exception e) {
+            log.error("\n Error while creating NewsCard node ");
+        }
+        return null;
+    }
+
+    private String addParentNode(Session session){
+        try {
+            if(session.nodeExists("/content/aemtraining/us/en")){
+                Node gParentNode=session.getNode("/content/aemtraining/us/en");
+                Node parentNode=gParentNode.addNode("us-newscard",NEWSCARDNODE_TYPE);
+                session.save();
+                return parentNode.getName();
+            }
+        }catch (Exception e){
+            log.error("\n Error while creating Parent node ");
+        }
+        return null;
+    }
+
+    private boolean saveImage(Node node, String link) {
+        try {
+            ResourceResolver resourceResolver = ResolverUtil.newResolver(resourceResolverFactory);
+
+            URL url = new URL(link);
+            URLConnection connection = url.openConnection();
```

Comment:
It's better to extract the image retrieval into a separate method, or an utility class. Because you see, it is not related to the logic of this class and even to the name of the current method. The method says "save" but a part of what you are doing is not "save" but "get from a remote endpoint".

Another reason for splitting into a "retrieval" method and a "saving" method is to differentiate exceptions. Asking for an image from internet can fail -- and then you don't even need to start saving anything. And, again, an exception saying "error when trying to download image..." and an exception saying "error when tryin to save image" -- are two completely different errors than need different treatment.